src/04-extensions.js

```diff
@@ -0,0 +1,212 @@
+const exampleDinosaurData = require("../data/dinosaurs");
+const exampleTicketData = require("../data/tickets");
+const exampleRoomData = require("../data/rooms");
+
+/*
+Write a function that will organize all of the dinosaurs by their period key. 
+The function should return an object where the keys are the period values, and 
+the values are an array of IDs.
+*/
+
+/* Expected Result
+  {
+    "Late Jurassic": [
+      "YLtkN9R37",
+      "GGvO1X9Zeh",
+      "BFjjLjea-O",
+      "iOVNUcv-ww",
+      "V53DvdhV2A",
+    ],
+    "Late Cretaceous": [
+      "WHQcpcOj0G",
+      "GKl035EYKN",
+      "2GglUqKT0G",
+      "Pr6kc4Q_Xf",
+      "ft5Gs5izdq",
+      "wuL4ddBinQ",
+    ],
+    // ...
+  }
+*/
+
+function organizeDinosaursByPeriod(dinos, key) {
+  //input an array of objects
+  //output an object where key represents period and value reps an array of dinoIDs
+
+  //create helper function to camelCase
+  function camelCaseKey(keyString) {
+    //create an array that contains all valid keys
+    const validKeys = ["name", "pronunciation", "meaningOfName", "diet", "lengthInMeters", "mya", "info"];
```

Comment:
In general, this is great. You are building a validation function and separating the keys that are valid.

However, this function is named `camelCaseKey()`. It seems like this function does two things: validates the keys _and_ sets it to camel case, in the event that it wasn't?

A good rule to keep in mind when writing functions is that a function should have a single responsibility. [There's even a Wikipedia page on it!](https://en.wikipedia.org/wiki/Single-responsibility_principle) So, I would have two functions here: `keysAreValid()` and `camelCase()`. Each one should do one thing.

For complex functions (e.g. `organizeDinosaursByPeriod()`), treat the overarching function as a sort of conductor. Its job is to call a bunch of other, smaller functions.